cnf-tests/testsuites/e2esuite/security/tuning_sysctl.go

```diff
@@ -0,0 +1,179 @@
+			podDefinition2 = pods.RedefineWithRestartPolicy(podDefinition2, corev1.RestartPolicyNever)
+			podDefinition2.Spec.NodeName = pod.Spec.NodeName
+			pod2, err := client.Client.Pods(TestNamespace).Create(context.Background(), podDefinition2, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			err = pods.WaitForPhase(client.Client, pod2, corev1.PodSucceeded, 1*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	Context("tuningcni over sriov", func() {
+		execute.BeforeAll(func() {
+			if discovery.Enabled() {
+				Skip("Tuned sriov tests disabled for discovery mode")
+			}
+			networks.CleanSriov(sriovclient, TestNamespace)
+			tuningConfig := sysctlConfig(map[string]string{Sysctl: "1"})
+			networks.CreateSriovPolicyAndNetwork(sriovclient, namespaces.SRIOVOperator, "test-network", "testresource", tuningConfig)
+		})
+
+		It("pods with sysctl's over sriov interface should start", func() {
+			podDefinition := pods.DefineWithNetworks(TestNamespace, []string{fmt.Sprintf("%s/%s", namespaces.SRIOVOperator, "test-network")})
+			pod, err := client.Client.Pods(TestNamespace).Create(context.Background(), podDefinition, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			err = pods.WaitForCondition(client.Client, pod, corev1.ContainersReady, corev1.ConditionTrue, 1*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("pods with sysctl's on bond over sriov interfaces should start", func() {
+			// NOTE: due to a bond cni bug we need to specify the name of the bond interface in both the cni config and
+			// the multus annotation for the network
+			bondLinkName := "bond0"
+			err := createBondNetworkAttachmentDefinitionWithSysctls(apiclient, TestNamespace, "bond", bondLinkName, "net1", "net2", map[string]string{Sysctl: "1"})
+			Expect(err).ToNot(HaveOccurred())
+			podDefinition := pods.DefineWithNetworks(TestNamespace, []string{
+				fmt.Sprintf("%s/%s", namespaces.SRIOVOperator, "test-network"),
+				fmt.Sprintf("%s/%s", namespaces.SRIOVOperator, "test-network"),
+				fmt.Sprintf("%s/%s@%s", TestNamespace, "bond", bondLinkName),
+			})
+			pod, err := client.Client.Pods(TestNamespace).Create(context.Background(), podDefinition, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			err = pods.WaitForCondition(client.Client, pod, corev1.ContainersReady, corev1.ConditionTrue, 1*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+	})
+	Context("clean SRIOV configuration", func() {
+		It("should restore the cluster to the original state", func() {
+			if !discovery.Enabled() {
+				By("cleaning the sriov test configuration")
+				networks.CleanSriov(sriovclient, TestNamespace)
+			}
+			err := namespaces.Delete(TestNamespace, apiclient)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+})
+
+func ceateMacVlanNetworkAttachmentDefinition(cs *client.ClientSet, namespace, nadName, ip string, sysctls map[string]string) error {
```

Comment:
I would:
- add a package "networkattachmentdefinition" to collect the functions there, or reuse the networks one
- have functions to create the definition, and then call networks.CreateNetworkAttachmentDefinition instead of hiding it inside
- have builder-like functions, like
macvlanConfig = WithSysctl(macvlanConfigFor(...), sysctlParameters)

Reply:
done
using the network package